command/catalog_nodes.go

```diff
@@ -0,0 +1,143 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/configutil"
+)
+
+// CatalogNodesCommand is a Command implementation that is used to fetch all the
+// datacenters the agent knows about.
+type CatalogNodesCommand struct {
+	BaseCommand
+}
+
+func (c *CatalogNodesCommand) Help() string {
+	helpText := `
+Usage: consul catalog datacenters [options]
+
+  Retrieves the list nodes registered in a given datacenter. By default, the
+  datacenter of the local agent is queried.
+
+  To retrieve the list of nodes:
+
+      $ consul catalog nodes
+
+  For a full list of options and examples, please see the Consul documentation.
+
+` + c.BaseCommand.Help()
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *CatalogNodesCommand) Run(args []string) int {
```

Comment:
Having it show tabular format similar to `members` (option 2) could still feed into `cut` or some other utility, but I don't really have a preference here, as long as there's a `-detailed` option.